router/cache/schema_cache.go

```diff
@@ -0,0 +1,102 @@
+package cache
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"sync"
+
+	"github.com/jackc/pgx/v5/pgproto3"
+	"github.com/pg-sharding/spqr/pkg/config"
+	"github.com/pg-sharding/spqr/pkg/models/kr"
+	"github.com/pg-sharding/spqr/pkg/pool"
+	"github.com/pg-sharding/spqr/pkg/startup"
+)
+
+type SchemaCache struct {
+	tableColumnsCache sync.Map
+	shardMapping      map[string]*config.Shard
+
+	pool *pool.DBPool
+}
+
+func NewSchemaCache(shardMapping map[string]*config.Shard) *SchemaCache {
+	var preferAZ string
+	if config.RouterConfig().PreferSameAvailabilityZone {
+		preferAZ = config.RouterConfig().AvailabilityZone
+	}
+	return &SchemaCache{
+		shardMapping:      shardMapping,
+		pool:              pool.NewDBPool(shardMapping, &startup.StartupParams{}, preferAZ),
+		tableColumnsCache: sync.Map{},
+	}
+}
+
+func (c *SchemaCache) GetColumns(be *config.BackendRule, schemaName, tableName string) ([]string, error) {
```

Comment:
GetColumns should use its own berule, defined in config. We can also default this berule to first defined berule, if not configured. I not sure what are consequences of resetting  berule here every time.